hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/compaction/log/CompactionFileInfo.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.hadoop.util.Preconditions;
+
+import java.util.Objects;
+
+/**
+ * Dao to keep SST file information in the compaction log.
+ */
+public final class CompactionFileInfo {
+  private final String fileName;
+  private final String startKey;
+  private final String endKey;
+  private final String columnFamily;
+
+  private CompactionFileInfo(String fileName,
+                             String startRange,
+                             String endRange,
+                             String columnFamily) {
+    this.fileName = fileName;
+    this.startKey = startRange;
+    this.endKey = endRange;
+    this.columnFamily = columnFamily;
+  }
+
+  public String getFileName() {
+    return fileName;
+  }
+
+  public String getStartKey() {
+    return startKey;
+  }
+
+  public String getEndKey() {
+    return endKey;
+  }
+
+  public String getColumnFamily() {
+    return columnFamily;
+  }
+
+  public HddsProtos.CompactionFileInfoProto getProtobuf() {
+    HddsProtos.CompactionFileInfoProto.Builder builder =
+        HddsProtos.CompactionFileInfoProto.newBuilder()
+            .setFileName(fileName);
+    if (startKey != null) {
+      builder = builder.setStartKey(startKey);
+    }
+    if (endKey != null) {
+      builder = builder.setEndKey(endKey);
+    }
+    if (columnFamily != null) {
+      builder = builder.setColumnFamily(columnFamily);
+    }
+    return builder.build();
+  }
+
+  public static CompactionFileInfo getFromProtobuf(
+      HddsProtos.CompactionFileInfoProto proto) {
+    Builder builder = new Builder(proto.getFileName());
+
+    if (proto.hasStartKey()) {
+      builder.setStartRange(proto.getStartKey());
+    }
+    if (proto.hasEndKey()) {
+      builder.setEndRange(proto.getEndKey());
+    }
+    if (proto.hasColumnFamily()) {
+      builder.setColumnFamily(proto.getColumnFamily());
+    }
+
+    return builder.build();
+  }
+
+  @Override
+  public String toString() {
+    return String.format("fileName: '%s', startKey: '%s', endKey: '%s'," +
+        " columnFamily: '%s'", fileName, startKey, endKey, columnFamily);
+  }
+
+  /**
+   * Builder of CompactionFileInfo.
+   */
+  public static class Builder {
+    private final String fileName;
+    private String startRange;
+    private String endRange;
+    private String columnFamily;
+
+    public Builder(String fileName) {
+      Preconditions.checkNotNull(fileName, "FileName is required parameter.");
+      this.fileName = fileName;
+    }
+
+    public Builder setStartRange(String startRange) {
+      this.startRange = startRange;
+      return this;
+    }
+
+    public Builder setEndRange(String endRange) {
+      this.endRange = endRange;
+      return this;
+    }
+
+    public Builder setColumnFamily(String columnFamily) {
+      this.columnFamily = columnFamily;
+      return this;
+    }
+
+    public CompactionFileInfo build() {
+      if ((startRange != null || endRange != null || columnFamily != null) &&
```

Comment:
Sorry, I'm confused about this condition. Either all three should be set or none of them should be set? Is that right?

Reply:
Yes, you are right.
Details: https://github.com/apache/ozone/pull/5303#discussion_r1327851516

Reply:
yeah